python-backend/app/api/users/minespace/resources/minespace_user.py

```diff
@@ -0,0 +1,53 @@
+import uuid
+
+from flask_restplus import Resource, reqparse
+from flask import request
+from app.extensions import api
+from ....utils.access_decorators import requires_role_mine_admin
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+from ..models.minespace_user import MinespaceUser
+from ..models.minespace_user_mine import MinespaceUserMine
+from app.extensions import db
+
+
+class MinespaceUserResource(Resource, UserMixin, ErrorMixin):
+    parser = reqparse.RequestParser()
+    parser.add_argument('email', type=str, location='json', required=True)
+    parser.add_argument('mine_guids', type=list, location='json', required=True)
+
+    @api.doc(params={'user_id': 'User id.'})
+    @requires_role_mine_admin
+    def get(self, user_id=None):
+        if user_id or request.args.get('email'):
+            user = MinespaceUser.find_by_id(user_id)
+            if not user:
```

Comment:
I think this would be easier to read if the email check was not nested inside of the id check.

If you just check for and `user_id` first, you can check `if not user and email` after without needing to nest them. I think this should give you the same behaviour.

And then after that, the `if not user` -> not found works the same, but all of the nesting is removed.

Reply:
Explored this suggestion but I prefer the current version of code for readability. Let me know if you want to revisit this though.